src/server/process_queue.ts

```diff
@@ -0,0 +1,11 @@
+import * as Pq from "p-queue"
+
+export abstract class ProcessQueue {
+    queue: Pq
+
+    constructor() {
+        this.queue = new Pq({concurrency: 1})
+    }
+
+    abstract async run(req: Express.Request, res: Express.Response): Promise<void>
```

Comment:
I really don't like having this interface involve Express. 

We should have some kind of base job which just knows how to fork a process, send arbitrary data across, and receive data back.

Then on top of that we can build a specialized job which takes an ActionRequest, sends it across to the proper place, performs the execute, and returns the ActionResponse back.